### PR TITLE
Canvas: Add ability to rotate elements

### PR DESCRIPTION
**Pull request by nmarrs**
More relatively low hanging fruit similar to https://github.com/grafana/grafana/pull/80407

Add ability to rotate canvas elements via the rotatable functionality that already [exists inside the moveable library](https://daybrush.com/moveable/storybook/index.html?path=/story/advanced-rotatable--advanced-rotatable-rotate-with-resize). This was previously [implemented temporarily](https://github.com/grafana/grafana/commit/588314b78eae011c89b469706ce968eff7ecfd84#diff-1b6b7f4e1606d259615982862325d5f98f6871ff486cc913803a313a1918a313) for the cancad hackathon project

https://github.com/grafana/grafana/assets/22381771/69f64108-7040-4059-aecb-6d66154970f8

TODO

- [x] Fix strange resizing / reset of rotation 
- [x] Test (all element types)
- [x] Improve handling of hiding custom ables
- [ ] merge (profit)

Known issues / next steps
- https://github.com/grafana/grafana/issues/84956
- https://github.com/grafana/grafana/issues/84957
- https://github.com/grafana/grafana/issues/84958
- https://github.com/grafana/grafana/issues/84959
- https://github.com/grafana/grafana/issues/84960
- https://github.com/grafana/grafana/issues/84961

### CODE REVIEW

**Comment by lukasztyrala** [comment]
Few comments:
* we should keep the cursor consistent when rotating
* it seems that moving an object will reset its rotation – I think it is not an anticipated behavior, as here the system, without warning, overwrites the previous intentional interaction by the user
* can we rotate by a fixed radius that would allow people to rotate by 90° or 45° easily? Or snap to certain positions (like 0°, 45°, 90°, …) – free rotation precision is tough to achieve with a mouse :-)

File: public/app/features/canvas/runtime/element.tsx [hunk 1]
```diff
@@ -432,16 +474,36 @@ export class ElementState implements LayerElement {
...
+
+    // Handle case when element is rotated
+    if (placement.rotation) {
+      const rotation = placement.rotation ?? 0;
+      const rotationInRadians = (rotation * Math.PI) / 180;
+      const originalDirLR = dirLR;
+      const originalDirTB = dirTB;
+
+      dirLR = Math.sign(originalDirLR * Math.cos(rotationInRadians) - originalDirTB * Math.sin(rotationInRadians));
+      dirTB = Math.sign(originalDirLR * Math.sin(rotationInRadians) + originalDirTB * Math.cos(rotationInRadians));
```
Comment (nmarrs) [inline public/app/features/canvas/runtime/element.tsx:505]: and you thought you were done with geometry in school? 🎒

Comment (baldm0mma) [reply public/app/features/canvas/runtime/element.tsx:505]: SOHCAHTOA!!!!

File: public/app/features/canvas/runtime/element.tsx [hunk 2]
```diff
@@ -99,6 +99,7 @@ export class ElementState implements LayerElement {
       // Minimum element size is 10x10
       minWidth: '10px',
       minHeight: '10px',
+      rotate: `${placement.rotation ?? 0}deg`,
```
[the remark this answers is not in the document]
Comment (baldm0mma) [reply public/app/features/canvas/runtime/element.tsx:102]: Will `placement.rotation` ever be nullish here? Don't you give it a default value above?

**Review by codeincarnate — APPROVED** [review]
Tested out in a number of different scenarios and seems to be working well. Good stuff! 😄

**Review by baldm0mma — APPROVED** [review]
Nice! Works as expected! I will say the snapping to other elements, especially if there are A LOT of elements feels a little jarring. But that's maybe not something we can easily change, since it is the expected behavior.

https://github.com/grafana/grafana/assets/46619047/d89e1c15-dba3-4b90-98cb-e28a0fb9721e

File: public/app/features/canvas/runtime/scene.tsx [hunk 3]
```diff
@@ -385,6 +385,22 @@ export class Scene {
...
+      settingsViewable: false,
+    };
+  };
+
+  enableCustomables = () => {
+    this.moveable!.props = {
+      dimensionViewable: true,
+      constraintViewable: true,
+      settingsViewable: true,
+    };
```
Comment (drew08t) [inline public/app/features/canvas/runtime/scene.tsx:401]: +100

**Review by drew08t — APPROVED** [review]

**Pull request merged**